### ISSUE TITLE
No module named nmap

### ISSUE THREAD

**Issue by Blackwolf499** [k4m4/kickthemout#358]
<!--
  Hi there! Thank you for bringing an issue to out attention.

- [ x] I have carefully read the [README](https://github.com/k4m4/kickthemout/blob/master/README.rst) file and haven't managed to resolve my issue.
- [ x] I have searched the [issues](https://github.com/k4m4/kickthemout/issues?utf8=%E2%9C%93&q=is%3Aissue) of this repo and believe that this is not a duplicate.
- [ x] I am running the latest version of KickThemOut.

- Hello again,

I've just re-installed kali yet again, but have come across yet another issue and the other issues don't seem to have a resolve. 

sudo apt-get update && sudo apt-get install nmap
Reading package lists... Done
Reading package lists... Done
Building dependency tree       
Reading state information... Done
nmap is already the newest version (7.70+dfsg1-3kali1).
0 upgraded, 0 newly installed, 0 to remove and 0 not upgraded.

It also states that all requirements have been satisfied but when i excute the script

sudo python3 kickthemout.py 

I get this 

ERROR: Requirements have not been satisfied properly. Please look at the README file for configuration instructions.

If you still cannot resolve this error, please submit an issue here:
	https://github.com/k4m4/kickthemout/issues

Details: No module named 'nmap'

Even though It states that I've satisfied all other requirements and I've already gone through the process of configuring all of them as well.

Regards,

**Comment by Blackwolf499** [comment]
ffs I just wrote an entire page detailing my error and now it's not showing.

**Comment by Blackwolf499** [comment]
Alright so let me try again, I've recently had an error but that was fixed but this is an entirely new error. 

sudo apt-get update && sudo apt-get install nmap
Reading package lists... Done
Reading package lists... Done
Building dependency tree       
Reading state information... Done
nmap is already the newest version (7.70+dfsg1-3kali1).
0 upgraded, 0 newly installed, 0 to remove and 0 not upgraded.

I already have nmap installed 
and when I execute this

sudo -H pip install -r requirements.txt (doing sudo -H pip3 install -r requirements.txt comes with an error)

I'm shown this 

Requirement already satisfied: kamene in /usr/local/lib/python2.7/dist-packages (from -r requirements.txt (line 1))
Requirement already satisfied: python-nmap in /usr/local/lib/python2.7/dist-packages (from -r requirements.txt (line 2))
Requirement already satisfied: netifaces in /usr/local/lib/python2.7/dist-packages (from -r requirements.txt (line 3))

But yet when I execute: Sudo python3 kickthemout.py this appears

ERROR: Requirements have not been satisfied properly. Please look at the README file for configuration instructions.

If you still cannot resolve this error, please submit an issue here:
	https://github.com/k4m4/kickthemout/issues

Details: No module named 'nmap'

Regards,

**Comment by k4m4** [comment]
The problem here is that when you do `pip install`, you are using pip for python2 instead of for python3. It is the error cause by `sudo -H pip3 install -r requirements.txt` that you must resolve.

**Comment by Blackwolf499** [comment]
Will look into it thanks.

**Comment by Blackwolf499** [comment]
OR: Requirements have not been satisfied properly. Please look at the README file for configuration instructions.

If you still cannot resolve this error, please submit an issue here:
	https://github.com/k4m4/kickthemout/issues

Details: No module named 'zlib'

**Comment by Blackwolf499** [comment]
I' ve installed zlib

**Comment by k4m4** [comment]
And even after installing zlib, you get a `No module named 'zlib'` error?

**Comment by Blackwolf499** [comment]
Yes that's correct

**Comment by Blackwolf499** [comment]
Would re-installing kickthemout.py resolve the issue?

**Comment by k4m4** [comment]
Hi, did you try re-installing?

**Comment by DwiWardana** [comment]
i think peoples get many error to running this tool because installing requirment using command "sudo -H pip3 install requirment", i using this command too, and i got many error something like "no module named nmap", "module nmap has no atribute PortScanner", "no module named kamene" and then i try to installing without -H, and it's worked now

**Comment by k4m4** [comment]
Closing due to inactivity. If you are still encountering a problem, feel free to re-open this issue. Thanks.
